easybuild/easyconfigs/t/Triton/Triton-3.1.0-foss-2024a-CUDA-12.6.0.eb

```
@@ -0,0 +1,88 @@
+# Update 3.1.0: Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2024/12
+# TODO: run unittests
+easyblock = 'PythonBundle'
+
+name = 'Triton'
+version = '3.1.0'
+versionsuffix = '-CUDA-%(cudaver)s'
+# There is no 3.1 in pypi and no 3.1-tag at github. However, 5fe38ffd is version bump 3.1 in the release_3.1.x branch:
+local_commit = '5fe38ffd73c2ac6ed6323b554205186696631c6f'
+# Triton 3.1.0 commit 5fe38ffd73c2ac6ed6323b554205186696631c6f requires a particular commit of LLVM 
+# (acc. to cmake/llvm-hash.txt: 10dc3a8e916d73291269e5e2b82dd22681489aa1, 2024/05/23):
+local_clang_commit = '10dc3a8e916d73291269e5e2b82dd22681489aa1'
+
+homepage = 'https://triton-lang.org/'
+
+description = """Triton is a language and compiler for parallel programming. It aims to provide a
+Python-based programming environment for productively writing custom DNN compute
+kernels capable of running at maximal throughput on modern GPU hardware."""
+
+toolchain = {'name': 'foss', 'version': '2024a'}
+
+github_account = 'openai'
+
+builddependencies = [
+    ('Clang', '18.1.6_%s' % local_clang_commit[:7], versionsuffix),
+    ('CMake', '3.29.3'),
+    ('Ninja', '1.12.1'),
+    ('pybind11', '2.13.6'),
+    ('poetry', '1.8.3'),
+    ('nlohmann_json', '3.11.3'),
+    ('googletest', '1.15.2'),
+]
+
+dependencies = [
+    ('CUDA', '12.6.0', '', SYSTEM),
+    ('Python', '3.12.3'),
+    ('Z3', '4.13.0'),
+]
+
+use_pip = True
+download_dep_fail = True
+
+_tr_start_dir = 'python'
+
+_tr_preinstallopts = 'export PYBIND11_SYSPATH=$EBROOTPYBIND11 && '
+_tr_preinstallopts += 'export JSON_SYSPATH=$EBROOTNLOHMANN_JSON && '
+_tr_preinstallopts += 'export LLVM_INCLUDE_DIRS=$EBROOTCLANG/include && '
+_tr_preinstallopts += 'export LLVM_LIBRARY_DIR=$EBROOTCLANG/lib && '
+_tr_preinstallopts += 'export LLVM_SYSPATH=$EBROOTCLANG && '
+_tr_preinstallopts += 'export TRITON_BUILD_WITH_CLANG_LLD=1 && '
+_tr_preinstallopts += 'export TRITON_HOME=/tmp/triton_home/ && '
```

we should try to avoid hardcoding `/tmp` just in case:
```suggestion
_tr_preinstallopts += 'export TRITON_HOME=${TMPDIR:-/tmp}/triton_home/ && '
```

---

makes sense :+1: